_posts/pl/2021-12-XX-transactions.md

```diff
@@ -0,0 +1,274 @@
+
+- **I** (_ang. isolation_) – izolacja – właściwość, która odseparowuje od siebie operacje wykonywane jednocześnie tak, aby te nie miały na siebie wpływu. Chodzi np. o sytuację, gdy odwołujemy się do tej samej tabeli w bazie. Można też powiedzieć, że izolacja zapewnia, że w przypadku wykonywania operacji w sposób równoległy baza danych pozostanie w takim stanie, jakby operacje te zostały wykonane sekwencyjnie.
+
+- **D** (_ang. durability_) – trwałość – gwarancja, że zaakceptowana transakcja nie zniknie niespodziewanie z systemu np. w przypadku awarii. Trwałość uzyskuje się poprzez zastosowanie mechanizmu replikacji, odnotowywania operacji w dzienniku logów czy poprzez zapis danych na dysku.
+
+### Poziomy izolacji
+W praktyce wiele transakcji wykonywanych jest równolegle przez System Zarządzania Bazą Danych. Oznacza to często równloegły dostęp aplikacji do tych samych danych w bazie. Warto zatem wspomnieć o sposobie ustawiania blokad w dostępie do danych. Sposób ustawiania blokad określany jest mianem **poziomu izolacji** transakcji i pomaga w zachowaniu integralności danych. Poziomy izolacji określają zasady **równoległej** realizacji kilku transakcji.
+
+System Zarządania Bazą Danych (_SZBD_) może np. zablokować wiersz tabeli, dopóki nie zostaną zatwierdzone aktualizacje. Efektem takiej blokady byłoby uniemożliwienie użytkownikowi uzyskania **brudnego odczytu**  (_ang. dirty read_). Odczyt tego typu polega na dostępie do zaktualizowanej wartości, która nie została jeszcze zatwierdzona, i której istnieje możliwość przywrócenia do wartości poprzedniej.
+
+**Niepowtarzalny odczyt** (_ang. non-repeatable read_) ma miejsce wtedy, gdy transakcja A pobiera wiersz, transakcja B go aktualizuje, a następnie transakcja A pobiera ten sam wiersz ponownie. W takim przypadku transakcja A pobiera dwa razy ten sam wiersz, ale widzi różne dane.
+
+**Odczyt fantomowy** (_ang. phantom read_) to sytuacja, gdy transakcja A pobiera zbiór wierszy będących w danym stanie. Transakcja B następnie wstawia lub aktualizuje wiersz do tabeli. Następnie transakcja A ponownie odczytuję zbiór wierszy widząc tym razem nowy lub zaktualizowany wiersz, określany jako fantom.
+
+Zgodnie ze standardem _SQL-92_ definiuje się 4 poziomy izolacji transakcji. Poniżej przedstawiono "efekty uboczne", które są akceptowalne na danym poziomie, poczynając od najmniej restrykcyjnego:
+
+| Poziom izolacji                | Brudne         | Niepowtarzalne         | Fantomowe         |
+|                                | odczyty        | odczyty                | odczyty           |
+| ------------------------------ |:--------------:|:----------------------:|:-----------------:|
+| `TRANSACTION_READ_UNCOMMITTED` | Dozwolone      | Dozwolone              | Dozwolone         |
```

Comment:
to chyba poziomy izolacji z jdbc. a w standardzie jest
- READ UNCOMMITTED,
- READ COMMITTED
- REPEATABLE READ
- SERIALIZABLE.